### PR TITLE
RED-180951 RED-181297 RED-184457 RED-184456 RED-184458 RED-184459 fixing bugs and improving the code

### PR DESCRIPTION
**Pull request by AvivDavid23**
Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Medium Risk**
> Touches RDB load paths for multiple data structures (Bloom, Cuckoo, CMS, TopK, TDigest), so mistakes could cause failed restores or memory handling regressions. Changes are mostly defensive validation with added tests, lowering security/stability risk from malformed payloads.
> 
> **Overview**
> Adds **defensive validation during RDB deserialization** across RedisBloom data structures to reject malformed payloads instead of risking overflows/OOB reads/writes. This includes stricter `bloom_validate_integrity` checks (`n2` bounds), Bloom RDB buffer-size and integrity validation, zero/invalid guard checks for Cuckoo `numFilters/numBuckets`, CMS array length validation, TDigest merged-nodes bounds checking, and TopK size/overflow checks replacing `assert`s (including pointer nulling on partial loads).
> 
> Updates tests to cover **malicious/corrupt `RESTORE` payloads** and memory/argument edge cases for CMS/Cuckoo/TopK/TDigest, and bumps module patch version to `2.6.28`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f4c0975b09b429439b4d2077df7fc3bdfe284072. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/rm_topk.c [hunk 1]
```diff
@@ -313,18 +313,46 @@ static void *TopKRdbLoad(RedisModuleIO *io, int encver) {
...
+        if (topk->heap) {
+            uint32_t buckets = heapSize / sizeof(HeapBucket);
+            buckets = buckets < topk->k ? buckets : topk->k;
+            for (uint32_t i = 0; i < buckets; ++i) {
+                topk->heap[i].item = NULL;
+            }
         }
+        err = true;
+        return NULL;
+    }
```
[the remark this answers is not in the document]
Comment (gabsow) [reply src/rm_topk.c:351]: We will take a look into it

File: src/rm_cms.c [hunk 2]
```diff
@@ -314,9 +314,21 @@ void *CMSRdbLoad(RedisModuleIO *io, int encver) {
...
     cms->depth = LoadUnsigned_IOError(io, err, NULL);
     cms->counter = LoadUnsigned_IOError(io, err, NULL);
-    size_t length = cms->width * cms->depth * sizeof(size_t);
+
+    if (cms->width == 0 || cms->depth == 0) {
+        err = true;
+        return NULL;
+    }
+
+    size_t expected_length = sizeof *cms->array * cms->width * cms->depth;
```
[the remark this answers is not in the document]
Comment (gabsow) [reply src/rm_cms.c:323]: We will take a look into it
